builder/vsphere/common/step_export.go

```diff
@@ -0,0 +1,295 @@
+//go:generate struct-markdown
+//go:generate mapstructure-to-hcl2 -type ExportConfig
+
+package common
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"fmt"
+	"hash"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/packer/builder/vsphere/driver"
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/template/interpolate"
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi/nfc"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// You may optionally export an ovf from VSphere to the instance running Packer.
+//
+// Example usage:
+//
+// ```json
+// ...
+//   "vm_name": "example-ubuntu",
+// ...
+//   "export": {
+//     "force": true,
+//     "output_directory": "./output_vsphere"
+//   },
+// ```
+// The above configuration would create the following files:
+//
+// ```
+// ./output_vsphere/example-ubuntu-disk-0.vmdk
+// ./output_vsphere/example-ubuntu.ovf
+// ```
+type ExportConfig struct {
+	// name of the ovf. defaults to the name of the VM
+	Name string `mapstructure:"name"`
+	// overwrite ovf if it exists
+	Force bool `mapstructure:"force"`
+	// include iso and img image files that are attached to the VM
+	Images bool `mapstructure:"images"`
+	// generate manifest using SHA 1, 256, 512. use 0 (default) for no manifest
+	Sha       int          `mapstructure:"sha"`
```

Comment:
What do you think about renaming the option to `sha_manifest` (underlying name `ShaManifest`) and changing this to a type string with the available options being "", "sha1", "sha256", "sha512"? 

The idea is to keep the sha1 selection types consistent with some of the other builders, and to be a bit more explicit that it will be used for generating a manifest.

Reply:
I think that changing it from Sha to Manifest (instead of ShaManifest) would be more clear. I'll change it to a string as the "0" makes the logic a little weird.